addons/website_payment/models/res_config_settings.py

```diff
@@ -53,6 +55,17 @@ def action_activate_stripe(self):
         menu_id = menu and menu.id
         return self.env.company._run_payment_onboarding_step(menu_id=menu_id)
 
+    def action_activate_razorpay(self):
+        self.ensure_one()
+        razorpay = self.env['payment.provider'].search([
+            *self.env['payment.provider']._check_company_domain(self.env.company),
+            ('code', '=', 'razorpay')
+        ], limit=1)
+
+        if not self.is_razorpay_supported_currency:
+            return False
+        return razorpay.action_razorpay_connect_account()
```

Comment:
```suggestion
         if not self.is_razorpay_supported_currency:
            return False

        razorpay = self.env['payment.provider'].search([
            *self.env['payment.provider']._check_company_domain(self.env.company),
            ('code', '=', 'razorpay')
        ], limit=1)
        return razorpay.action_razorpay_connect_account()
```

When `is_razorpay_supported_currency` is false, razorpay var is not unused.